core/src/main/resources/hudson/tasks/Messages.properties

```diff
@@ -36,7 +36,7 @@ ArtifactArchiver.NoIncludes=\
 No artifacts are configured for archiving.\n\
 You probably forgot to set the file pattern, so please go back to the configuration and specify it.\n\
 If you really did mean to archive all the files in the workspace, please specify "**"
-ArtifactArchiver.NoMatchFound=No artifacts found that match the file pattern "{0}". Configuration error?
+ArtifactArchiver.NoMatchFound=No artifacts found that match the file pattern "{0}". Did you mean to use the "allowEmptyArchive" option?
```

Comment:
Why does this not make sense in freestyle builds? Even if the problem is not the consequence of an earlier failure, you might simply have wanted to allow zero artifacts in some cases.